src/main/scala/nodes/images/Convolver.scala

```diff
@@ -101,29 +101,31 @@ object Convolver {
       whitener: Option[ZCAWhitener],
       varConstant: Double): DenseMatrix[Double] = {
     var x,y,chan,pox,poy,py,px = 0
-    while (x < resWidth) {
-      y = 0
-      while (y < resHeight) {
-        chan = 0
-        while (chan < imgChannels) {
-          poy = 0
-          while (poy < convSize) {
-            pox = 0
-            while (pox < convSize) {
+
+    poy = 0
+    while (poy < convSize) {
```

Comment:
the function call might get inlined in our case, but who knows.

There's also this: https://github.com/nativelibs4java/Scalaxy/tree/master/Loops

Which is the macro-rewriting optimized for loop thing that someone else has done for us. I might try both - but after getting decent convolver unit tests put together.